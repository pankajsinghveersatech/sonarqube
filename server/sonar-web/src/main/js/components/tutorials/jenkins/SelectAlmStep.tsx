/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import RadioToggle from '../../../components/controls/RadioToggle';
import AlertSuccessIcon from '../../../components/icons/AlertSuccessIcon';
import { hasMessage, translate } from '../../../helpers/l10n';
import { AlmKeys } from '../../../types/alm-settings';
import Step from '../components/Step';

export interface SelectAlmStepProps {
  alm?: AlmKeys;
  open: boolean;
  onCheck: (value: AlmKeys) => void;
  onOpen: () => void;
}

function getAlmLongName(alm: AlmKeys) {
  return hasMessage('alm', alm, 'long') ? translate('alm', alm, 'long') : translate('alm', alm);
}

export default function SelectAlmStep(props: SelectAlmStepProps) {
  const { alm, open } = props;
  return (
    <Step
      finished={true}
      open={open}
      onOpen={props.onOpen}
      renderForm={() => (
        <div className="boxed-group-inner">
          <RadioToggle
            name="alm"
            onCheck={props.onCheck}
            options={[
              AlmKeys.BitbucketCloud,
              AlmKeys.BitbucketServer,
              AlmKeys.GitHub,
              AlmKeys.GitLab
            ].map(almKey => ({
              label: getAlmLongName(almKey),
              value: almKey
            }))}
            value={alm}
          />
        </div>
      )}
      renderResult={() =>
        alm && (
          <div className="boxed-group-actions display-flex-center">
            <AlertSuccessIcon className="spacer-right" />
            {getAlmLongName(alm)}
          </div>
        )
      }
      stepTitle={translate('onboarding.tutorial.with.jenkins.alm_selection.title')}
    />
  );
}
