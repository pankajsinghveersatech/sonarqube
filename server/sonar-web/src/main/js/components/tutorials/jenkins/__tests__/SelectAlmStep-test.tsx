/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { shallow } from 'enzyme';
import * as React from 'react';
import { AlmKeys } from '../../../../types/alm-settings';
import Step from '../../components/Step';
import SelectAlmStep, { SelectAlmStepProps } from '../SelectAlmStep';

jest.mock('../../../../helpers/l10n', () => ({
  hasMessage: (_a: string, k: string, _b: string) => k === AlmKeys.BitbucketCloud,
  translate: (...k: string[]) => k.join('.')
}));

it('should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot('default');
  expect(
    shallowRender()
      .find(Step)
      .props()
      .renderForm()
  ).toMatchSnapshot('form, default');
  expect(
    shallowRender({ alm: AlmKeys.Azure })
      .find(Step)
      .props()
      .renderForm()
  ).toMatchSnapshot('form, with alm');
  expect(
    shallowRender()
      .find(Step)
      .props().renderResult!()
  ).toBeUndefined();
  expect(
    shallowRender({ alm: AlmKeys.BitbucketCloud })
      .find(Step)
      .props().renderResult!()
  ).toMatchSnapshot('result, with alm');
});

function shallowRender(props: Partial<SelectAlmStepProps> = {}) {
  return shallow<SelectAlmStepProps>(
    <SelectAlmStep onCheck={jest.fn()} onOpen={jest.fn()} open={true} {...props} />
  );
}
