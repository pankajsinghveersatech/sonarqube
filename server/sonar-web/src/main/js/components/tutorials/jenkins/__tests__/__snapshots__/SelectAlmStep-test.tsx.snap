// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly: default 1`] = `
<Step
  finished={true}
  onOpen={[MockFunction]}
  open={true}
  renderForm={[Function]}
  renderResult={[Function]}
  stepTitle="onboarding.tutorial.with.jenkins.alm_selection.title"
/>
`;

exports[`should render correctly: form, default 1`] = `
<div
  className="boxed-group-inner"
>
  <RadioToggle
    disabled={false}
    name="alm"
    onCheck={[MockFunction]}
    options={
      Array [
        Object {
          "label": "alm.bitbucketcloud.long",
          "value": "bitbucketcloud",
        },
        Object {
          "label": "alm.bitbucket",
          "value": "bitbucket",
        },
        Object {
          "label": "alm.github",
          "value": "github",
        },
        Object {
          "label": "alm.gitlab",
          "value": "gitlab",
        },
      ]
    }
    value={null}
  />
</div>
`;

exports[`should render correctly: form, with alm 1`] = `
<div
  className="boxed-group-inner"
>
  <RadioToggle
    disabled={false}
    name="alm"
    onCheck={[MockFunction]}
    options={
      Array [
        Object {
          "label": "alm.bitbucketcloud.long",
          "value": "bitbucketcloud",
        },
        Object {
          "label": "alm.bitbucket",
          "value": "bitbucket",
        },
        Object {
          "label": "alm.github",
          "value": "github",
        },
        Object {
          "label": "alm.gitlab",
          "value": "gitlab",
        },
      ]
    }
    value="azure"
  />
</div>
`;

exports[`should render correctly: result, with alm 1`] = `
<div
  className="boxed-group-actions display-flex-center"
>
  <AlertSuccessIcon
    className="spacer-right"
  />
  alm.bitbucketcloud.long
</div>
`;
